internal/configs/configmaps.go

```diff
@@ -501,6 +509,33 @@ func ParseConfigMap(cfgm *v1.ConfigMap, nginxPlus bool, hasAppProtect bool) *Con
 		}
 	}
 
+	if hasAppProtectDos {
+
+		appProtectDosLivenessEnable, livenessExistsEnable, errEnable := GetMapKeyAsBool(cfgm.Data, "app-protect-dos-liveness-enable", cfgm)
+		appProtectDosLivenessURI, livenessExistsURI := cfgm.Data["app-protect-dos-liveness-uri"]
+		appProtectDosLivenessPort, livenessExistsPort, errPort := GetMapKeyAsInt(cfgm.Data, "app-protect-dos-liveness-port", cfgm)
+
+		if livenessExistsEnable {
+			if errEnable != nil {
+				glog.Error(errEnable)
+			} else if errPort != nil {
+				glog.Error(errPort)
+			} else {
+				if appProtectDosLivenessEnable {
+					cfgParams.MainAppProtectDosLivenessEnable = "on"
```

Comment:
I'm not sure about use a variable to mean two different things. It appears in places where variables are set to 'on' or 'off'.

It's probably not a good thing as the variable has two different meanings, one for enabling/disabling the value and one for the value itself 'on/off'. 

Can we simplify the logic and and set the default value to 'off' ? Then, if `appProtectDosLivenessEnable` is true, we can set it to 'on' ?

Reply:
Make enable default value "off", and change the logic.